src/main/java/org/spongepowered/api/map/color/MapColors.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map.color;
+
+import org.spongepowered.api.map.util.MapColorFactory;
+import org.spongepowered.api.map.util.MapColorMatcher;
+import org.spongepowered.api.util.Color;
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+
+import java.util.Collection;
+import java.util.function.Function;
+
+/**
+ * An enumeration of all the possible instances given {@link MapColor.Base}
+ */
+public final class MapColors {
+
+    private static final MapColorFactory factory = DummyObjectProvider.createFor(MapColorFactory.class, "factory");
```

Comment:
It's not really a factory, it's just as you said, poor way to abstract out the implementation hidden details into a class. I've been trying to chip away at it, but since those colors aren't a `CatalogType` I still don't have a way of eliminating that last method in a nice clean way. I could stuff it in `Registry` I suppose, that's where it was in the past. But that seemed odd.